modules/cells/src/main/java/dmg/cells/nucleus/SystemCell.java

```diff
@@ -66,15 +51,15 @@
                 _packetsReplied,
                 _exceptionCounter;
    private final Runtime                _runtime = Runtime.getRuntime() ;
-   private final Gate                   _shutdownLock = new Gate(false);
+   private final Semaphore              _shutdownLock = new Semaphore(0);
 
    private class TheKiller extends Thread {
       @Override
       public void run(){
          _log.info("Running shutdown sequence");
          kill() ;
          _log.info("Kill done, waiting for shutdown lock");
-         _shutdownLock.check() ;
+         _shutdownLock.tryAcquire() ;
```

Comment:
you break the semantics here: Gate#check blocks until gate is open. Semaphore#tryAcquire returns right away with true/false result.